solarforecastarbiter/io/fetch/arm.py

```diff
@@ -174,18 +191,20 @@ def fetch_arm(user_id, api_key, datastreams, start, end):
         requested period.
     """
     site_dfs = []
-    for ds, ds_type in datastreams.items():
-        fns = list_arm_filenames(user_id, api_key, ds, start, end)
-        var_dfs = []
-        for fn in fns:
-            file = retrieve_arm_dataset(user_id, api_key, fn)
-            # TODO: allow users to define ds_type and applicable variables
-            if ds_type == 'irrad':
-                var = extract_arm_variable(file, 'down_short_diffuse_hemisp')
-            elif ds_type == 'weather':
-                var = extract_arm_variable(file, 'temp_mean')
-            var_dfs.append(var)
-            file.close()
-        site_dfs.append(pd.concat(var_dfs, sort=False))
-    new_data = pd.concat(site_dfs, axis=1)
-    return new_data
+    fns = list_arm_filenames(user_id, api_key, datastream, start, end)
+    for fn in fns:
+        nc_file = retrieve_arm_dataset(user_id, api_key, fn)
```

Comment:
you're explicitly calling `nc_file.close()` below, which makes we want to see everything from here until there in a context manager or a `try/except/finally` block.

Reply:
That's a good point. I think we can get away just omitting the nc_file.cose() call  because there is no real file handle. For some reason the Dataset constructor requires a filename argument even when using the memory option.